File: trikot-viewmodels-declarative-flow/sample/ios/Sample/SampleImageProvider.swift
import SwiftUI
import Jasper
import Trikot

class SampleImageProvider: VMDImageProvider {
    func imageForResource(imageResource: VMDImageResource) -> Image? {
        guard let resource = imageResource as? SampleImageResource else { return nil }
        switch resource {
        case .iconClose:
            return Image("icn_close")
        case .imageBridge:
            return Image("bridge")
        case .imagePlaceholder:
            return Image("placeholder")
        default:
            return nil
        }
    }
}

File: trikot-viewmodels-declarative-flow/sample/android/src/main/java/com/mirego/sample/resource/SampleImageProvider.kt
package com.mirego.sample.resource

import android.content.Context
import com.mirego.sample.resources.SampleImageResource
import com.mirego.trikot.viewmodels.declarative.configuration.VMDImageProvider
import com.mirego.trikot.viewmodels.declarative.properties.VMDImageResource
import com.mirego.trikot.viewmodels.declarative.properties.VMDNoImageResource
import com.mirego.vmd.sample.android.R

class SampleImageProvider : VMDImageProvider {
    override fun resourceIdForResource(resource: VMDImageResource, context: Context): Int? {
        return when (resource) {
            is VMDNoImageResource -> null
            is SampleImageResource -> mapSampleImageResource(resource)
            else -> null
        }
    }

    private fun mapSampleImageResource(resource: VMDImageResource): Int? {
        return when (resource) {
            SampleImageResource.ICON_CLOSE -> R.drawable.ic_close
            SampleImageResource.IMAGE_BRIDGE -> R.drawable.bridge
            SampleImageResource.IMAGE_PLACEHOLDER -> R.drawable.placeholder
            else -> null
        }
    }
}
